javascript/modules/client/src/client.ts

```diff
@@ -576,7 +621,30 @@ export class Client extends ClientCore implements IClient {
       data: encodeWithdrawActionData(params)
     }
   }
-  private async _buildUpdateMetadataAction(daoAddreessOrEns: string, _params: IMetadata): Promise<DaoAction> {
+
+  private async _buildUpdateMetadataAction(daoAddreessOrEns: string, ipfsUri: string): Promise<DaoAction> {
+    // check ens
+    let address = daoAddreessOrEns
+    if (!isAddress(daoAddreessOrEns)) {
+      const resolvedAddress = await this.web3.getSigner()?.resolveName(daoAddreessOrEns)
+      if (!resolvedAddress) {
+        throw new Error("invalid ens")
+      }
+      address = resolvedAddress
+    }
+    const ipfsRegex = /^ipfs:\/\/(Qm[1-9A-HJ-NP-Za-km-z]{44,}|b[A-Za-z2-7]{58,}|B[A-Z2-7]{58,}|z[1-9A-HJ-NP-Za-km-z]{48,}|F[0-9A-F]{50,})/gm
```

Comment:
Shouldn't this have a line end symbol in the end `$`? Otherwise, the `split()` below may catch extra garbage being given after the CiD